ntp-proto/src/time_types.rs

```diff
@@ -180,6 +180,19 @@ impl NtpDuration {
         )
     }
 
+    /// Interpret an exponent `k` as `2^k` seconds, expressed as an NtpDuration
+    pub(crate) fn from_exponent(input: i8) -> Self {
+        let seconds = if input < 0 {
+            // 1. / (1L << -(a))
+            1.0 / (1i64 << -input) as f64
+        } else {
+            // 1L << (a)
+            (1i64 << input) as f64
+        };
```

Comment:
Style nit: I'd write this as:

```rust
Self::from_seconds(match input {
    // 1L << (a)
    0.. => (1i64 << input) as f64,
    // 1. / (1L << -(a))
    _ => 1.0 / (1i64 << -input) as f64,
})
```

Reply:
actually, do we expect problems with just using 

```
Self::from_seconds(2.0f64.powi(input as i32))
```

My tests don't see any precision loss (makes sense given it's a power of 2) and this code expresses what we're trying to achieve much better than the bitshifting.

Reply:
Hah, that looks much nicer, and I think it makes sense.